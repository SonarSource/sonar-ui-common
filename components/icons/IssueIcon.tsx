/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import BugIcon from './BugIcon';
import VulnerabilityIcon from './VulnerabilityIcon';
import CodeSmellIcon from './CodeSmellIcon';
import SecurityHotspotIcon from './SecurityHotspotIcon';

interface Props {
  className?: string;
  type: T.IssueType;
  size?: number;
}

export default function IssueIcon({ className, type, size }: Props) {
  switch (type) {
    case 'BUG':
      return <BugIcon className={className} size={size} />;
    case 'VULNERABILITY':
      return <VulnerabilityIcon className={className} size={size} />;
    case 'CODE_SMELL':
      return <CodeSmellIcon className={className} size={size} />;
    case 'SECURITY_HOTSPOT':
      return <SecurityHotspotIcon className={className} size={size} />;
    default:
      return null;
  }
}
