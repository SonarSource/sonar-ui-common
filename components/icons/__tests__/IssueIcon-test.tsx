/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import IssueIcon from '../IssueIcon';

it('should render correctly', () => {
  expect(shallowRender('BUG')).toMatchSnapshot();
  expect(shallowRender('VULNERABILITY')).toMatchSnapshot();
  expect(shallowRender('CODE_SMELL')).toMatchSnapshot();
  expect(shallowRender('SECURITY_HOTSPOT')).toMatchSnapshot();
});

function shallowRender(type: T.IssueType) {
  return shallow(<IssueIcon type={type} />);
}
