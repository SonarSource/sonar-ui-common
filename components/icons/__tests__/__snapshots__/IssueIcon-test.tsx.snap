// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `<BugIcon />`;

exports[`should render correctly 2`] = `<VulnerabilityIcon />`;

exports[`should render correctly 3`] = `<CodeSmellIcon />`;

exports[`should render correctly 4`] = `<SecurityHotspotIcon />`;
